sdk/core/Azure.Core.Experimental/src/DynamicJson.cs

```diff
@@ -0,0 +1,588 @@
+
+
+        public static explicit operator bool?(DynamicJson json) => json._kind == JsonValueKind.Null ? (bool?)null : json.GetBoolean();
+        public static explicit operator int?(DynamicJson json) => json._kind == JsonValueKind.Null ? (int?)null : json.GetIn32();
+        public static explicit operator long?(DynamicJson json) => json._kind == JsonValueKind.Null ? (long?)null : json.GetLong();
+        public static explicit operator float?(DynamicJson json) => json._kind == JsonValueKind.Null ? (float?)null : json.GetFloat();
+        public static explicit operator double?(DynamicJson json) => json._kind == JsonValueKind.Null ? (double?)null : json.GetDouble();
+
+        public static implicit operator DynamicJson(int value) => new DynamicJson(value);
+        public static implicit operator DynamicJson(long value) => new DynamicJson(value);
+        public static implicit operator DynamicJson(double value) => new DynamicJson(value);
+        public static implicit operator DynamicJson(float value) => new DynamicJson(value);
+        public static implicit operator DynamicJson(bool value) => new DynamicJson(value);
+        public static implicit operator DynamicJson(string? value) => new DynamicJson((object?)value);
+        public static implicit operator DynamicJson(int? value) => new DynamicJson(value);
+        public static implicit operator DynamicJson(long? value) => new DynamicJson(value);
+        public static implicit operator DynamicJson(double? value) => new DynamicJson(value);
+        public static implicit operator DynamicJson(float? value) => new DynamicJson(value);
+        public static implicit operator DynamicJson(bool? value) => new DynamicJson(value);
+
+        public string? GetString() => (string?)EnsureValue();
+
+        public int GetIn32()
+        {
+            var value = EnsureNumberValue().AsLong();
+            if (value > int.MaxValue || value < int.MinValue)
+            {
+                throw new OverflowException();
+            }
+            return (int)value;
+        }
+
+        public long GetLong() => EnsureNumberValue().AsLong();
+        public float GetFloat()
+        {
+            var value = EnsureNumberValue().AsDouble();
+            if (value > float.MaxValue || value < float.MinValue)
+            {
+                throw new OverflowException();
+            }
+            return (float)value;
+        }
+        public double GetDouble() => EnsureNumberValue().AsDouble();
+        public bool GetBoolean() => (bool)EnsureValue()!;
+        public int GetArrayLength() => EnsureArray().Count;
+        public DynamicJson GetProperty(string name) => GetPropertyValue(name);
+
+        public static DynamicJson Serialize<T>(T value, JsonSerializerOptions? options = null)
+        {
+            var serialized = JsonSerializer.Serialize<T>(value, options);
+            return new DynamicJson(JsonDocument.Parse(serialized).RootElement);
```

Comment:
`JsonDocument` is `IDisposable`.

Who is responsible for disposing it? If you don't, APIs like these can accidentally exhaust the underlying array pool.

For your specific case, consider cloning the element to detach it from the document's lifetime, and dispose the document.
https://docs.microsoft.com/en-us/dotnet/api/system.text.json.jsonelement.clone?view=netcore-3.1
Something like this:
```C#
using doc = JsonDocument.Parse(serialized);
return new DynamicJson(doc.RootElement.Clone());
```